### DISCUSSION TITLE
Restart the app

### DISCUSSION THREAD

**Discussion by Shahin-rmz** [zauberzeug/nicegui#1719] [Q&A]
### Question

Hello,
Today after just 2 and a half days of work, I have finished a web app, which I will show to my colleages.
That's why `NiceGUI` is impressive.
For small refinement, I have a question.

Is there is way to `restart` the app after using `ui.shutdown`? 
I want to the user be able to kill a process and work with the app again, if wanted.

**Comment by falkoschindler** [comment]
Hi @Shahin-rmz, that's great to hear! 😀

There's currently no built-in way to restart a NiceGUI app. But in some of our projects we do something like this:
```py
ui.button('restart app', on_click=lambda: os.utime('main.py'))
```
This updates the modification time of the main.py file. Assuming you use `ui.run(..., restart=True)` (the default), this will trigger a restart since main.py seems to have changed.

**Reply by Shahin-rmz** [reply]
That's awesome. 
I'll try it out.
Thanks.

**Reply by Aues6uen11Z** [reply]
This approach isn't perfect. For example, if I use `asyncio.create_subprocess_exec` in my program, I can't set `reload=True`(https://github.com/zauberzeug/nicegui/issues/486)
😢
